Source/DiabloUI/progress.cpp

```diff
@@ -71,16 +92,29 @@ void ProgressRender(BYTE progress)
 
 bool UiProgressDialog(int (*fnfunc)())
 {
-	ProgressLoad();
 	SetFadeLevel(256);
 
+	// Blit the background once and then free it.
+	ProgressLoadBackground();
+	ProgressRenderBackground();
+	if (RenderDirectlyToOutputSurface && IsDoubleBuffered()) {
+		// Blit twice for triple buffering.

```

Comment:
Isn't twice only enough for double buffering?

Reply:
This is twice after the initial blit -- so 3 times in total